apps/script/HoC2023ScriptFiles/HoC2023AiGenerateWeights.py

```diff
@@ -0,0 +1,78 @@
+# Please refer to the README.md file in this directory for more information
+# on how to run this script to generate the cached association maps used by the client.
+
+import json
+from HoC2023AiHelperFunctions import *
+from constants import *
+
+emoji_model_descriptive_names, emojis_map = get_ai_emoji_inputs()
+
+background_effect_blockly_ids, background_effect_user_facing_names = get_background_effects()
+palette_blockly_ids, palette_user_facing_names = get_palettes()
+foreground_effect_blockly_ids, foreground_effect_user_facing_names = get_foreground_effects()
+
+blockly_ids = {
+    PALETTES: palette_blockly_ids,
+    BACKGROUND_EFFECTS: background_effect_blockly_ids,
+    FOREGROUND_EFFECTS: foreground_effect_blockly_ids
+}
+
+# Below, we store them in maps for which the key is the model_descriptive_name and the
+# value is the corresponding blockly_id.
+model_descriptive_names = {}
```

Comment:
Nit: don't need to initialize with empty dictionary?

Reply:
I had tried not initializing previously, but received an error about the dictionary not being defined yet.